### PR TITLE
get and set display name for XOpenDisplay()

### PR DESCRIPTION
**Pull request by jub3i**
can be done by just setting DISPLAY env var, but nice if you want more control then you can now set it programmatically. ran into this problem when using ssh.

### CODE REVIEW

File: src/robotjs.cc [hunk 1]
```diff
@@ -472,6 +474,26 @@ NAN_METHOD(getScreenSize)
...
+	NanReturnValue(NanNew<String>(getXDisplay()));
+}
+
+NAN_METHOD(setXDisplayName)
+{
+	NanScope();
+
+	//Convert arg to c-string
+	//NOTE: surely better way to go from v8::String to char* ?
+	std::string name =
```
Comment (jub3i) [inline src/robotjs.cc]: this feels pretty dirty, you know better way to accomplish?

Comment (jub3i) [inline src/robotjs.cc]: this feels pretty dirty, you know better way to accomplish?

File: src/robotjs.cc [hunk 2]
```diff
@@ -2,12 +2,14 @@
...
 #include <v8.h>
 #include <vector>
+#include "string.h"
 #include "mouse.h"
 #include "deadbeef_rand.h"
 #include "keypress.h"
 #include "screen.h"
 #include "screengrab.h"
 #include "MMBitmap.h"
+#include "xdisplay.h"
```
Comment (octalmage) [inline src/robotjs.cc]: This will probably cause errors on other OSs, right?

**Comment by octalmage** [comment]
If I understand correctly this allows you to pick the X server you connect to?

File: src/robotjs.cc (hunk 1 above)
Comment (octalmage) [inline src/robotjs.cc]: This should work! 
```
NanUtf8String name(arg[0]);
char *display_name = *name;
```

**Comment by jub3i** [comment]
Thanks that argument parsing worked nicely. I fixed up the OS compat issue, although I'm not sure I like how it structured, thought about maybe moving some of it into `screen.h` to avoid having to use those defines in `robotjs.cc`.

Yes, that’s exactly the purpose of this code; directly from X(7) under the DISPLAY NAMES section: 
```
From the user's perspective, every X server has a display name of the form: 

hostname:displaynumber.screennumber

This information is used by the application to determine how it should connect to the server and which screen it should use by default (on displays with multiple monitors)...
```

**Comment by octalmage** [comment]
Sorry for the delay, I've been without internet and I want to test this first. I should be able to test tonight or tomorrow.

**Comment by octalmage** [comment]
#206 replaces this.

**Pull request closed without merging**